### ISSUE TITLE
Bug: issue when migrating to the new `groups` for `sort-object-types` / `sort-object-interfacs`

### ISSUE THREAD

**Issue by OlivierZal** [azat-io/eslint-plugin-perfectionist#409]
### Describe the bug

You can see in the commit below that despite the following `groups`:
```ts
  groups: [
    'required-index-signature',
    'optional-index-signature',
    'required-property',
    'optional-property',
    'required-method',
    'optional-method',
  ],
```

- `onSync?` is moved ahead of required methods, although it's an optional method: [here](https://github.com/OlivierZal/melcloud-api/commit/777fce9deea2018089209b00a86d3833bd568ecf#diff-e0899f9bf34e50cd996d55c63b887d2fbd99d1b9cf18264928709c28f33700baR86)
- `tiles` is a required method but is moved ahead of all the other required methods: every occurrence, e.g. [here](https://github.com/OlivierZal/melcloud-api/commit/777fce9deea2018089209b00a86d3833bd568ecf#diff-caf90ffb40aa6dfd6a5b98f50a7b43825066d1eafc25be1b2a4ba7103cca40cdR48)

It seems they are both considered to be properties instead of methods, probably because:
- the first is typed with an alias (`OnSyncFunction`)
- the second includes an intersection in its type

### Code example

https://github.com/OlivierZal/melcloud-api/commit/777fce9deea2018089209b00a86d3833bd568ecf

### ESLint version

v9.16.0

### ESLint Plugin Perfectionist version

v4.2.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by OlivierZal** [comment]
@hugop95, FYI.

**Comment by hugop95** [comment]
@OlivierZal 

> It seems they are both considered to be properties instead of methods, probably because:
> - the first is typed with an alias (OnSyncFunction)
> - the second includes an intersection in its type

Your deduction is correct: functions are detected only under the form `a: (x) => y` or `a(x): y`.

> - the first is typed with an alias (OnSyncFunction)

Aliases are not parsed because the plugin doesn't do any sort of type check. Not much we can do at the moment.

> the second includes an intersection in its type

That can theoretically be fixed, assuming that we have no type alias, we can recursively check if a type union/intersection is a function.

**Comment by azat-io** [comment]
Added in ac1e7c4b5168ecab0028c0880f0fd750e95f3441.
Released in [v4.3.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.3.0).

**Comment by hugop95** [comment]
Update regarding

> Aliases are not parsed because the plugin doesn't do any sort of type check. Not much we can do at the moment.

Early POC regarding type detection is promising, we might have a fix for that soon.
